typescript/cli/src/send/transfer.ts

```diff
@@ -179,74 +130,77 @@ async function executeDelivery({
   const provider = multiProvider.getProvider(origin);
   const connectedSigner = signer.connect(provider);
 
-  // TODO replace all code below with WarpCore
-  // https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3259
+  const warpCore = WarpCore.FromConfig(
+    MultiProtocolProvider.fromMultiProvider(multiProvider),
+    warpCoreConfig,
+  );
 
-  if (tokenType === TokenType.collateral) {
-    const wrappedToken = await getWrappedToken(routerAddress, provider);
-    const token = ERC20__factory.connect(wrappedToken, connectedSigner);
-    const approval = await token.allowance(signerAddress, routerAddress);
-    if (approval.lt(wei)) {
-      const approveTx = await token.approve(routerAddress, wei);
-      await approveTx.wait();
+  if (!routerAddress) {
+    const tokensForRoute = warpCore.getTokensForRoute(origin, destination);
+    if (tokensForRoute.length === 0) {
+      logRed(`No Warp Routes found from ${origin} to ${destination}`);
+      return;
     }
+
+    routerAddress = (await select({
+      message: `Select router address`,
+      choices: [
+        ...tokensForRoute.map((t) => ({
+          value: t.addressOrDenom,
```

Comment:
It would be good to add a `description` field here with more info about the token, such as symbol and/or name: https://github.com/SBoudrias/Inquirer.js/tree/master/packages/select#usage